### DISCUSSION TITLE
Cluster Stopped Working on Different Wifi

### DISCUSSION THREAD

**Discussion by oheckmann** [k3s-io/k3s#6505] [Q&A]
Situation: I have a five node RPI Cluster connected with K3s (Specifically a PicoCluster 5H with 5 RPI4 8GB) in high Availability mode with embedded DB, 3 master nodes, and 2 worker nodes. The whole system has been working fine in my home, on my home Wi-Fi. I was under the impression that each of the RPI were locating each other through the use of their ethernet cables. For a project, wanted to demonstrate what I was doing on my cluster (some analytics) in a different setting. So I turned off my cluster and drove it to a different location. I sshed into each of the nodes and connected to the local Wi-Fi using raspi-config.

Problem: When I tried to do anything with the cluster from any of the Control nodes, I get the problem `Error from server (ServiceUnavailable): apiserver not ready
` 
When I check the logs with `systemctl status k3s`, get the following logs
`
Nov 15 16:34:38 master0 k3s[701]: {"level":"warn","ts":"2022-11-15T16:34:38.267Z","caller":"rafthttp/probing_status.go:68","msg":"prober detected unhealthy status","round-tripper-name":"ROUND_TRIPPER_RAFT_MESSA>
Nov 15 16:34:38 master0 k3s[701]: {"level":"info","ts":"2022-11-15T16:34:38.550Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"729eab7ef84f74c6 is starting a new election at term 146"}
Nov 15 16:34:38 master0 k3s[701]: {"level":"info","ts":"2022-11-15T16:34:38.550Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"729eab7ef84f74c6 became pre-candidate at term 146"}
Nov 15 16:34:38 master0 k3s[701]: {"level":"info","ts":"2022-11-15T16:34:38.550Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"729eab7ef84f74c6 received MsgPreVoteResp from 729eab7ef84f74c6 at te>
Nov 15 16:34:38 master0 k3s[701]: {"level":"info","ts":"2022-11-15T16:34:38.550Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"729eab7ef84f74c6 [logterm: 146, index: 17184346] sent MsgPreVote req>
Nov 15 16:34:38 master0 k3s[701]: {"level":"info","ts":"2022-11-15T16:34:38.550Z","logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"729eab7ef84f74c6 [logterm: 146, index: 17184346] sent MsgPreVote req>
Nov 15 16:34:38 master0 k3s[701]: time="2022-11-15T16:34:38Z" level=info msg="Waiting to retrieve kube-proxy configuration; server is not ready: https://127.0.0.1:6443/v1-k3s/readyz: 500 Internal Server Error"
Nov 15 16:34:40 master0 k3s[701]: {"level":"warn","ts":"2022-11-15T16:34:40.253Z","logger":"etcd-client","caller":"v3@v3.5.3-k3s1/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd>
Nov 15 16:34:40 master0 k3s[701]: time="2022-11-15T16:34:40Z" level=info msg="Failed to test data store connection: context deadline exceeded"
Nov 15 16:34:40 master0 k3s[701]: time="2022-11-15T16:34:40Z" level=info msg="Tunnel server egress proxy waiting for runtime core to become available"
log file: 3.5.3-k3s1/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcdESC
`
Am I correct in understanding that the problem is that I am on a different Wi-Fi, and that it will be resolved if I go back to my home Wi-Fi, provided that I simply do not mess with it? If this is the case, is there something I can do to fix the problem if it occurs in the future? If the problem is more complicated than that, what do I do?
I will provide any other logs that you need

**Comment by brandond** [comment]
Assuming your nodes get the same IP when you go back, yes. In the future, you might look at setting the --node-ip to the correct stable address.

**Reply by oheckmann** [reply]
They did do that. 
For the setting the  --node-ip to the correct stable address, I would just install reinstall K3s on each node in the cluster with the additional flag --node-ip <IP of Node>, and the cluster would continue to work, even if I move to a different wifi network. 
Am I understanding this correctly?
